pkg/agent/resourcemanager/fetcher/kubelet/topology/topology_adapter.go

```diff
@@ -315,6 +330,40 @@ func (p *topologyAdapterImpl) addNumaSocketChildrenZoneNodes(generator *util.Top
 	return nil
 }
 
+// addDeviceZoneNodes add the device nodes which are children of numa zone nodes to the generator, the device nodes are
+// generated by generateZoneNode according to TopologyLevel, Type and Name in TopologyAwareAllocatableQuantityList
+func (p *topologyAdapterImpl) addDeviceZoneNodes(generator *util.TopologyZoneGenerator,
+	allocatableResources *podresv1.AllocatableResourcesResponse) error {
+	if allocatableResources == nil {
+		return fmt.Errorf("allocatable Resources is nil")
+	}
+	var errList []error
+	for targetResourceName, targetZoneName := range p.resourceNameToZoneNameMap {
+		for _, device := range allocatableResources.Devices {
+			if targetResourceName == device.ResourceName {
+				for _, deviceId := range device.DeviceIds {
+					deviceNode := util.GenerateDeviceZoneNode(deviceId, targetZoneName)
+					if len(device.Topology.Nodes) == 0 {
+						continue
+					}
+					numaZoneNode := util.GenerateNumaZoneNode(int(device.Topology.Nodes[0].ID))
+					err := generator.AddNode(&numaZoneNode, deviceNode)
```

Comment:
range device.Topology.Nodes to add Zone instead of assuming one device topology match one numa node ?

Reply:
done